tests/catch/unit/memory/hipFree.cc

```diff
@@ -0,0 +1,580 @@
+#if HT_NVIDIA
+  HipTest::HIP_SKIP_TEST("EXSWCPHIPT-120");
+  return;
+#endif
+  size_t width = GENERATE(32, 512, 1024);
+  size_t height = GENERATE(0, 32, 512, 1024);
+  hipArray_t arrayPtr{};
+  hipExtent extent{};
+  extent.width = width;
+  extent.height = height;
+  hipChannelFormatDesc desc = hipCreateChannelDesc<char>();
+
+  HIP_CHECK(hipMallocArray(&arrayPtr, &desc, extent.width, extent.height, hipArrayDefault));
+
+  HIP_CHECK(freeStuff(arrayPtr, ArrayFree));
+  HIP_CHECK_ERROR(freeStuff(arrayPtr, ArrayFree), hipErrorContextIsDestroyed);
+}
+#if HT_NVIDIA
+TEST_CASE("Unit_hipFreeDoubleArrayDestroy") {
+#if HT_NVIDIA
+  HipTest::HIP_SKIP_TEST("EXSWCPHIPT-120");
+  return;
+#endif
+  size_t width = GENERATE(32, 512, 1024);
+  size_t height = GENERATE(0, 32, 512, 1024);
+  DriverContext ctx{};
+  // HipTest::HIP_SKIP_TEST("EXSWCPHIPT-81");
+  // return;
+
+  hiparray ArrayPtr{};
+  HIP_ARRAY_DESCRIPTOR cuDesc;
+  cuDesc.Width = width;
+  cuDesc.Height = height;
+  cuDesc.Format = HIP_AD_FORMAT_UNSIGNED_INT8;
+  cuDesc.NumChannels = 2;
+  HIP_CHECK(hipArrayCreate(&ArrayPtr, &cuDesc));
+  HIP_CHECK(freeStuff(ArrayPtr, ArrayDestroy));
+  HIP_CHECK_ERROR(freeStuff(ArrayPtr, ArrayDestroy), hipErrorContextIsDestroyed);
+}
+#endif
+
+
+// DevFree, ArrayFree, ArrayDestroy, HostFree Mutithreaded
+TEMPLATE_TEST_CASE("Unit_hipFreeMultiTDev", "", char, int, float2, float4) {
+  constexpr size_t numAllocs = 10;
+  std::vector<TestType*> ptrs(numAllocs);
+  size_t allocSize = sizeof(TestType) * GENERATE(1, 32, 64, 128);
+
+  for (auto& ptr : ptrs) {
+    HIP_CHECK(hipMalloc(&ptr, allocSize));
+  }
+
+  std::vector<std::thread> threads;
+
+  for (auto ptr : ptrs) {
+    threads.push_back(
+        std::thread([ptr]() { HIP_CHECK_THREAD(workIsDoneCheck<TestType*>(ptr, DevFree)); }));
+  }
+
+  for (auto& t : threads) {
+    t.join();
+  }
+  HIP_CHECK_THREAD_FINALIZE();
+}
+
+TEMPLATE_TEST_CASE("Unit_hipFreeMultiTHost", "", char, int, float2, float4) {
+  constexpr size_t numAllocs = 10;
+  std::vector<TestType*> ptrs(numAllocs);
+  size_t allocSize = sizeof(TestType) * GENERATE(1, 32, 64, 128);
+
+  for (auto& ptr : ptrs) {
+    HIP_CHECK(hipHostMalloc(&ptr, allocSize));
+  }
+
+  std::vector<std::thread> threads;
+
+  for (auto ptr : ptrs) {
+    threads.push_back(
+        std::thread([ptr]() { HIP_CHECK_THREAD(workIsDoneCheck<TestType*>(ptr, HostFree)); }));
+  }
+
+  for (auto& t : threads) {
+    t.join();
+  }
+  HIP_CHECK_THREAD_FINALIZE();
+}
+
+#if HT_NVIDIA
+TEMPLATE_TEST_CASE("Unit_hipFreeMultiTArray", "", char, int, float2, float4) {
+  constexpr size_t numAllocs = 10;
+  size_t width = GENERATE(32, 128, 256, 512, 1024);
+  size_t height = GENERATE(32, 128, 256, 512, 1024);
+  DriverContext ctx;
+
+
+  SECTION("ArrayDestroy") {
+    std::vector<hiparray> ptrs(numAllocs);
+    HIP_ARRAY_DESCRIPTOR cuDesc;
+    cuDesc.Width = width;
+    cuDesc.Height = height;
+    cuDesc.Format = HIP_AD_FORMAT_UNSIGNED_INT8;
+    cuDesc.NumChannels = 2;
```

Comment:
you've hard coded the layout here. there is a helper class for this [here](https://github.com/ROCm-Developer-Tools/HIP/blob/1c4822f24a99448c14f375ac9dd54cde2c598a54/tests/catch/unit/memory/hipArrayCommon.hh#L37) which will give you the correct format and numchannel at compile time. example use [here](https://github.com/ROCm-Developer-Tools/HIP/blob/1c4822f24a99448c14f375ac9dd54cde2c598a54/tests/catch/unit/memory/hipArrayCreate.cc#L224)

Reply:
Added vec_info in testing